### DISCUSSION TITLE
Pushing to TLM fifo in SystemVerilog from python

### DISCUSSION THREAD

**Discussion by msrasheed** [pyuvm/pyuvm#97] [Q&A] [answered]
So I already have a lot of UVM infrastructure written in SystemVerilog. I'd like to reuse it without having to reimplement everything in python. 
Is it possible to make TLM transactions between the python and SystemVerilog UVM parts?

**Comment by raysalemi** [comment, marked as answer]
> Is it possible to make TLM transactions between the python and SystemVerilog UVM parts?  

Siemens has released **pyquesta** on `Pypi.org`.  This allows you to pass transfer objects between Python and SystemVerilog using function calls.  You can put the objects into SystemVerilog `uvm_queue` and use a sequence to read them and create sequence items for QVIP. 

You can contact your Questa AE for the documentation.
